### PR TITLE
Fix KSLinux Support

### PR DESCRIPTION
**Pull request by Takeya-Yuki**
## Description
Fix Build for #726 and #724

### CODE REVIEW

**Comment by Takeya-Yuki** [comment]
Add KSLinux Support
Add KSLinux Logo
Fix #726

File: neofetch [hunk 1]
```diff
@@ -102,7 +102,8 @@ get_distro() {
 
             elif type -p tazpkg >/dev/null; then
                 distro="SliTaz $(< /etc/slitaz-release)"
-
+            elif type -p kpm > /dev/null; then
```
Comment (dylanaraps) [inline neofetch]: Add a newline above this.

File: neofetch [hunk 2]
```diff
@@ -434,6 +435,8 @@ get_packages() {
 
             type -p dpkg >/dev/null && \
                 packages="$((packages+=$(dpkg --get-selections | grep -cv deinstall$)))"
+            type -p kpm >/dev/null && \
```
Comment (dylanaraps) [inline neofetch]: Add a newline above this.

File: neofetch [hunk 3]
```diff
@@ -1633,8 +1636,13 @@ get_term() {
     # Check $PPID for terminal emulator.
     while [[ -z "$term" ]]; do
         parent="$(get_ppid "$parent")"
-        name="$(get_process_name "$parent")"
+        if [ "$parent" -lt 2 ]; then
```
Comment (dylanaraps) [inline neofetch]: What does this do and why is it needed?

File: neofetch [hunk 4]
```diff
@@ -3372,7 +3385,7 @@ get_distro_colors() {
             ascii_file="void"
         ;;
 
-        *"[Windows 10]"* | *"on Windows 10"* | "Windows 8"* | "Windows 10"* | "windows10" | "windows8" )
+        *"[Windows 10]"* | *"on Windows 10"* | "Windows 8"* | "Windows 10"*)
```
Comment (dylanaraps) [inline neofetch]: Add these back.

File: neofetch [hunk 5]
```diff
@@ -3067,6 +3075,11 @@ get_distro_colors() {
             ascii_file="korora"
         ;;
 
+        "KSLinux"*)
+            set_colors 4 7 1
+            ascii_file="ks"
```
Comment (dylanaraps) [inline neofetch]: Name the file "kslinux"

File: ascii/distro/ks [hunk 6]
```diff
@@ -0,0 +1,11 @@
+${c1} K   K U   U RRRR   ooo
```
Comment (dylanaraps) [inline ascii/distro/ks]: Name the file "kslinux"

File: neofetch (hunk 3 above)
Comment (Takeya-Yuki) [inline neofetch]: because shell is running as PID 1
(KSLinux has no init executable)

**Comment by Takeya-Yuki** [comment]
I Had Fix All Bugs , and Modified as your notice.
KSLinux has a built-in initrd, so the PID of bash is 1
It'll get into a dead loop without PID=1 Check

**Comment by dylanaraps** [comment]
Remove the parent check and instead make the loop look like this:

```sh
while [[ -z "$term" ]] || ((parent > 2)); do
```

**Comment by Takeya-Yuki** [comment]
parent Check script I added has been removed

File: neofetch [hunk 7]
```diff
@@ -1634,10 +1640,10 @@ get_term() {
     esac
 
     # Check $PPID for terminal emulator.
-    while [[ -z "$term" ]]; do
-        parent="$(get_ppid "$parent")"
-        name="$(get_process_name "$parent")"
+    while [[ -z "$term" ]] || ((parent > 1)); do
```
Comment (konimex) [inline neofetch]: I don't think the whitespace here is necessary.

Comment (Takeya-Yuki) [inline neofetch]: which whitespace?

**Comment by Takeya-Yuki** [comment]
Remove the parent check and instead make the loop look like this:

while [[ -z "$term" ]] || ((parent > 2)); do
This should fix it.
@dylanaraps 
This will not work properly due to *sh will not set $term value ($term will always blank, and run into a dead loop)
((parent > 2)) test command will not run.

File: neofetch (hunk 7 above)
Comment (konimex) [inline neofetch]: [This one](https://github.com/Takeya-Yuki/neofetch/blob/4b0cf233a86e150766cf6f45d18045a3e71a29d7/neofetch#L1644).

**Comment by dylanaraps** [comment]
Lets remove all of the `get_term` changes and we'll handle it in a separate PR.

**Comment by Takeya-Yuki** [comment]
I Had Removed all the changes of get_term()

**Pull request merged**